docs/i2k_tutorial.md

```diff
@@ -11,86 +11,86 @@ You can find more details about this tutorial here: [ImJoying Interactive Bioima
 
-?> If you want to share the definition of a specific api function to someone, you can simply click on the function name and copy the url in the address bar (e.g.: https://imjoy.io/docs/#/api?id=apialert).
+?> If you want to share the definition of a specific api function with someone, you can click on the function name and copy the url in the address bar (e.g.: https://imjoy.io/docs/#/api?id=apialert).
 
 You can directly access the `api` object in Javascript plugins (with type=`window` or `web-worker`):
 <!-- ImJoyPlugin: { "type": "web-worker", "passive": true,"editor_height": "200px"} -->
 ```js
 api.alert("Hello from ImJoy!")
 ```
 
-In Python plugins (type=`web-python` or `native-python`), you will need to do `from imjoy import api` before you can access the `api` object.
+In Python plugins (type=`web-python` or `native-python`), you will need to add `from imjoy import api` before you can access the `api` object.
 ```python
-# import it 
+# import api object
 from imjoy import api
 ...
-# use it 
+# use api object 
 api.alert("Hello from ImJoy!")
 ```
 
-?> In this tutorial we will show ImJoy API usage mostly in Javascript, but the Python code are similar except written in a different syntax.
+?> In this tutorial we will show ImJoy API usage mostly for Javascript, but the Python code will be similar except some smaller differences in the syntax.
 
 ### Remote Procedure Calls in ImJoy
 
-Although calling `alert()` and `api.alert()` appears the same, the underlying processes are different. While calling `alert()`, the popup dialog is initiated from the plugin, calling `api.alert()` means the popup dialog is initiated by the ImJoy core.
+Although calling `alert()` and `api.alert()` produces the same results (a popup message), it is important to understand that the underlying processes are different. When calling `alert()`, the popup dialog is initiated from the plugin directly, while calling `api.alert()` results in the ImJoy core initating the popup dialog.
 
-
-Keep in mind that ImJoy run each plugin in isolated or sandboxed environment (i.e. sandboxed iframe, webworker, conda virtual environment or docker container).
+Keep in mind that ImJoy run each plugin in and isolated or sandboxed environment (i.e. sandboxed iframe, webworker, conda virtual environment or docker container).
 
 When calling an ImJoy API function from a plugin, the function will be executed in the ImJoy core. The plugin runs in a different environment, all the functions defined in the ImJoy core are "remote" functions. In comparison, all the functions defined in the same plugin are "local".
 
-?> Therefore, calling ImJoy API functions means doing **Remote Procedure Calls (RPC)**.
+?> Therefore, calling ImJoy API functions means performing a **Remote Procedure Calls (RPC)**.
 
 ?> ImJoy supports bidirectional RPCs, not only between the plugin and the ImJoy core, but also between plugins. This can be consistently done across programming languages and hosting computers.
```

Comment:
Please fix this line: not only with the call but also between plugins

Reply:
? not sure If I understand what to fix ...

Reply:
Sorry, after read it again, it's fine.